### PR TITLE
fix: issue with custom styles piercing shadow DOM rendering

### PR DESCRIPTION
**Pull request by castastrophe**
tldr;

Content set had moved the tabs and accordion into the shadow DOM for controlled rendering but found that this strips all styles from the panels.  Not good!!  This patch reverts that change and keeps the tabs and accordion in the light DOM so that panel styles will continue to work.

### Preview

- [Content set preview](https://deploy-preview-1365--happy-galileo-ea79c4.netlify.app/elements/pfe-content-set/demo/) 

### Testing instructions

1. Regression testing on pfe-content-set
2. Add custom styles to the panel region and ensure those styles persist when the content set upgrades.

#### Browser requirements

Your component should work in all of the following environments:

- [x] Latest 2 versions of Edge
- [x] Internet Explorer 11 (should be useable, not pixel perfect)
- [x] Latest 2 versions of Firefox (one on Mac OS, one of Windows OS)
- [x] Firefox 68 (or latest version for Red Hat Enterprise Linux distribution)
- [x] Latest 2 versions of Chrome (one on Mac OS, one of Windows OS)
- [x] Latest 2 versions of Safari
- [x] Android mobile device (such as the Galaxy S9)
- [x] Apple mobile device (such as the iPhone X)
- [x] Apple tablet device (such as the iPhone Pro)

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one request or issue (no stragglers or scope-creep).
- [x] Tests have been updated to cover these changes.
- [x] Browser testing passed.
- [x] Repository compiles and tests pass.
- [x] Changelog updated.

### Merging

Please **squash** when merging and ensure your commit message uses [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) formatting.

**Be sure to share your updates with the [patternfly-elements-contribute@redhat.com](mailto:patternfly-elements-contribute@redhat.com) mailing list!**

### CODE REVIEW

File: elements/pfe-content-set/src/pfe-content-set.js [hunk 1]
```diff
@@ -133,6 +133,24 @@ class PfeContentSet extends PFElement {
     super.connectedCallback();
 
     if (this.hasLightDOM()) {
+      this.setAttribute("hidden");
```
Comment (heyMP) [inline elements/pfe-content-set/src/pfe-content-set.js:136]: This will set the `pfe-content-set` element to hidden.  Is this intended?

File: elements/pfe-content-set/src/pfe-content-set.js [hunk 2]
```diff
@@ -133,6 +133,24 @@ class PfeContentSet extends PFElement {
...
+
+      // If the tab does not exist in the light DOM, add it
+      if (!this.tabs) {
+        let newEl = document.createElement("pfe-tabs");
+        newEl.setAttribute("hidden", "");
```
Comment (heyMP) [inline elements/pfe-content-set/src/pfe-content-set.js:141]: So the theory is that we set everything to hidden and then let the `_toggleVisible` function add or remove those attributes. Do I have that correct? :)

File: elements/pfe-content-set/demo/index.html [hunk 3]
```diff
@@ -95,9 +95,12 @@ <h4 pfe-content-set--header>Heading 1</h4>
           varius blandit sit amet non magna. Etiam porta sem malesuada magna mollis euismod.</p>
       </div>
       <h4 pfe-content-set--header>Heading 2</h4>
-      <div pfe-content-set--panel>Content for heading 2. Sed posuere consectetur est at lobortis. Donec id elit non mi
+      <div pfe-content-set--panel>
```
Comment (heyMP) [inline elements/pfe-content-set/demo/index.html:98]: I don't think this is a breaking change but I wanted to double check.  Do `pfe-content-set` panels rely on its children to be wrapped in an element?  Can they still be text nodes?

**Review by heyMP — COMMENTED** [review]
Really creative use of the slots! 🤹

File: elements/pfe-content-set/demo/index.html (hunk 3 above)
Comment (castastrophe) [reply elements/pfe-content-set/demo/index.html:98]: They can be either. I added this to the demo page to confirm incoming styles were retained.

File: elements/pfe-content-set/src/pfe-content-set.js (hunk 2 above)
Comment (castastrophe) [reply elements/pfe-content-set/src/pfe-content-set.js:141]: Toggle visible is only impacting the visibility of the tab and accordion. The hidden on content set is to prevent the flash before upgrade of the light DOM data.

File: elements/pfe-content-set/demo/index.html (hunk 3 above)
Comment (castastrophe) [reply elements/pfe-content-set/demo/index.html:98]: (Oh and this is just the demo page)

File: elements/pfe-content-set/src/pfe-content-set.js [hunk 4]
```diff
@@ -132,11 +138,10 @@ class PfeContentSet extends PFElement {
   connectedCallback() {
     super.connectedCallback();
 
-    if (this.hasLightDOM()) {
-      Promise.all([customElements.whenDefined(PfeTabs.tag), customElements.whenDefined(PfeAccordion.tag)]).then(() => {
-        this._build();
-      });
-    }
+    // If the element has an ID, postfix container
+    if (this.id) this.id = `${this.id}-container`;
```
Comment (castastrophe) [inline elements/pfe-content-set/src/pfe-content-set.js:142]: @heyMP I was coding last night and remembered we moved the content set into the light DOM to resolve an issue with duplicate IDs so I added some logic to postfix the container and heading IDs.

File: elements/pfe-content-set/src/pfe-content-set.js [hunk 5]
```diff
@@ -150,6 +155,21 @@ class PfeContentSet extends PFElement {
     if (window.ResizeObserver) this._resizeObserver.disconnect();
   }
 
+  fetchComponent() {
```
Comment (castastrophe) [inline elements/pfe-content-set/src/pfe-content-set.js:158]: Due to the duplicate ID issue, I went back to the old approach of creating a new component and deleting the old one if the rendering mode changes.

File: elements/pfe-content-set/src/pfe-content-set.js [hunk 6]
```diff
@@ -111,17 +111,23 @@ class PfeContentSet extends PFElement {
     return this.parentNode ? this.parentNode.offsetWidth > breakpointValue : window.outerWidth > breakpointValue;
```
Comment (castastrophe) [inline elements/pfe-content-set/src/pfe-content-set.js:111]: TODO: What happens if I put something not pfe-tab or pfe-accordion in the rendered-component slot?

File: elements/pfe-content-set/src/pfe-content-set.html [hunk 7]
```diff
@@ -1,5 +1,2 @@
-<div id="lightdom" hidden>
-  <slot></slot>
-</div>
-<pfe-tabs hidden visible-at="large"></pfe-tabs>
-<pfe-accordion hidden visible-at="small"></pfe-accordion>
+<slot id="lightdom" hidden></slot>
+<slot name="rendered-component"></slot>
```
Comment (castastrophe) [inline elements/pfe-content-set/src/pfe-content-set.html:2]: Q: Comment or attribute indicating this is a private slot?

Comment (castastrophe) [inline elements/pfe-content-set/src/pfe-content-set.html:2]:
```suggestion
<slot name="_upgradedComponent"></slot>
```

Comment (castastrophe) [reply elements/pfe-content-set/src/pfe-content-set.html:2]: TODO: Update references in the js

File: elements/pfe-content-set/src/pfe-content-set.js [hunk 8]
```diff
@@ -111,17 +111,23 @@ class PfeContentSet extends PFElement {
...
-    return this.shadowRoot.querySelector(`${PfeAccordion.tag}[visible-at="small"]`);
+    return this.querySelector(`pfe-accordion[slot="rendered-component"]`);
   }
 
   constructor() {
     super(PfeContentSet);
 
+    this.fetchComponent = this.fetchComponent.bind(this);
+
     this._mutationHandler = this._mutationHandler.bind(this);
```
Comment (castastrophe) [inline elements/pfe-content-set/src/pfe-content-set.js:131]: Can we watch the slot with the mutation observer instead?  Mutation observer on the component is picking up when tabs & accordion update.

File: elements/pfe-content-set/src/pfe-content-set.html [hunk 9]
```diff
@@ -1,5 +1,2 @@
-<div id="lightdom" hidden>
-  <slot></slot>
-</div>
-<pfe-tabs hidden visible-at="large"></pfe-tabs>
-<pfe-accordion hidden visible-at="small"></pfe-accordion>
+<slot id="lightdom" hidden></slot>
+<slot name="_view" private></slot>
```
Comment (heyMP) [inline elements/pfe-content-set/src/pfe-content-set.html:2]: nice!  I like that.

**Review by heyMP — APPROVED** [review]

**Review by starryeyez024 — APPROVED** [review]
Life: Gift To Mankind

**Comment by castastrophe** [comment]
1 more test to fix: https://github.com/patternfly/patternfly-elements/runs/1922191577#step:8:2022
